Source/SwiftLintFramework/Rules/Style/NumberSeparatorRule.swift

```diff
@@ -137,24 +137,27 @@ public struct NumberSeparatorRule: OptInRule, CorrectableRule, ConfigurationProv
 
         let shouldAddSeparators = clean.count >= minimumLength
 
-        for (idx, char) in reversedIfNeeded(Array(clean), reversed: !isFraction).enumerated() {
+        var idx = 0
```

Comment:
Why is the `idx` parameter extracted into a manually incremented variable? As far as I can see, when the `guard` statements condition is `false`, then the `continue` will skip incrementing the `idx` variable. Is this intentional? And if so, why is this correct? Please either revert to the `enumerated()` variant or properly document the reasoning behind the manuel `idx` handling including why skipping the incrementation step prevents issues here.

Reply:
I added a comment. Let me know if this makes sense, or if you have a way which is more clear.

Reply:
Ahh, now I understand. I think the easiest way to make this clear would be simply to rename the genericly named var `idx` to something like `numerals`. Then I would have immediately understood even without the comment. I think, having a proper name makes the comment unecessary, too.

Reply:
Ah, okay. That makes sense and I completely agree. I'll update the PR.